### PR TITLE
OCTOPUS-650: Changes made to exclude creation of the Transit Gateway related activities by default

### PR DESCRIPTION
**Pull request by Chandan-Abhyankar**
setup_transit_gateway flag is added. It's default value false 

If setup_transit_gateway = false
User needs to manually create Transit Gateway , valid connections and pass it's name to 'transit_gateway_name' variable 

If setup_transit_gateway = true
Creation of Transit Gateway, it's connections will be taken care by automation. Also no need to specify 'transit_gateway_name' variable. Transit Gateway name will be constructed automatically based on VPC name.

### CODE REVIEW

File: modules/1_vpc_support/outputs.tf [hunk 1]
```diff
@@ -16,5 +16,13 @@ output "vpc_crn" {
 }
 
 output "transit_gateway_id" {
-  value = var.override_transit_gateway_name == "" ? module.transit_gateway[0].new_tg : module.existing_gateway[0].existing_tg
+  value = var.setup_transit_gateway == true ? module.transit_gateway[0].new_tg : module.existing_gateway[0].existing_tg
+}
+
+output "transit_gateway_name" {
+  value = var.setup_transit_gateway == false ? module.existing_gateway[0].existing_tg_name : module.transit_gateway[0].new_tg_name
```
Comment (prb112) [inline modules/1_vpc_support/outputs.tf:23]:
```suggestion
  value = !var.setup_transit_gateway ? module.existing_gateway[0].existing_tg_name : module.transit_gateway[0].new_tg_name
```

File: modules/1_vpc_support/outputs.tf [hunk 2]
```diff
@@ -16,5 +16,13 @@ output "vpc_crn" {
...
+}
+
+output "transit_gateway_status" {
+  value = var.setup_transit_gateway == false ? module.existing_gateway[0].existing_tg_status : module.transit_gateway[0].new_tg_status
```
Comment (prb112) [inline modules/1_vpc_support/outputs.tf:27]:
```suggestion
  value = !var.setup_transit_gateway ? module.existing_gateway[0].existing_tg_status : module.transit_gateway[0].new_tg_status
```

File: modules/1_vpc_support/support.tf [hunk 3]
```diff
@@ -52,32 +52,34 @@ module "security_groups" {
 }
 
 module "existing_gateway" {
-  count = var.override_transit_gateway_name != "" ? 1 : 0
+  count = var.setup_transit_gateway != true ? 1 : 0
```
Comment (prb112) [inline modules/1_vpc_support/support.tf:55]:
```suggestion
  count = !var.setup_transit_gateway ? 1 : 0
```

File: modules/1_vpc_support/support.tf [hunk 4]
```diff
@@ -52,32 +52,34 @@ module "security_groups" {
...
+  setup_transit_gateway = var.setup_transit_gateway
+  transit_gateway_name  = var.transit_gateway_name
+  resource_crn          = data.ibm_is_vpc.vpc.resource_crn
+  vpc_name              = var.vpc_name
+  mac_tags              = var.mac_tags
 }
 
 module "transit_gateway" {
-  count = var.override_transit_gateway_name != "" ? 0 : 1
+  count = var.setup_transit_gateway != true ? 0 : 1
```
Comment (prb112) [inline modules/1_vpc_support/support.tf:70]:
```suggestion
  count = !var.setup_transit_gateway ? 0 : 1
```

File: modules/1_vpc_support/variables.tf [hunk 5]
```diff
@@ -60,6 +60,7 @@ variable "vpc_supp_public_ip" {
 }
 
 variable "powervs_machine_cidr" {}
-variable "override_transit_gateway_name" {}
+variable "setup_transit_gateway" {}
```
Comment (prb112) [inline modules/1_vpc_support/variables.tf:63]: We could make this explicitly a boolean

File: modules/3_transit_gateway/transit_gateway.tf [hunk 6]
```diff
@@ -4,7 +4,7 @@
 ################################################################
 
 module "generated" {
-  count = var.override_transit_gateway_name == "" ? 1 : 0
+  count = var.setup_transit_gateway == true ? 1 : 0
```
Comment (prb112) [inline modules/3_transit_gateway/transit_gateway.tf:7]:
```suggestion
  count = var.setup_transit_gateway ? 1 : 0
```

**Review by prb112 — CHANGES_REQUESTED** [review]
I recommended some small changes to better follow the style. 
Looks good otherwise.

File: modules/1_vpc_support/outputs.tf [hunk 7]
```diff
@@ -16,5 +16,13 @@ output "vpc_crn" {
 }
 
 output "transit_gateway_id" {
-  value = var.override_transit_gateway_name == "" ? module.transit_gateway[0].new_tg : module.existing_gateway[0].existing_tg
+  value = var.setup_transit_gateway == true ? module.transit_gateway[0].new_tg : module.existing_gateway[0].existing_tg
```
[the remark this answers is not in the document]
Comment (Chandan-Abhyankar) [reply modules/1_vpc_support/outputs.tf:19]: Yes, setup_transit_gateway is Boolean. Accepting your suggested changes. Thanks

File: modules/1_vpc_support/variables.tf (hunk 5 above)
Comment (Chandan-Abhyankar) [reply modules/1_vpc_support/variables.tf:63]: It is defined as Boolean in main variables.tf

**Review by prb112 — APPROVED** [review]

**Pull request merged**